### PR TITLE
Added debug and debugChunks combinators

### LINKED ISSUE #1569: `showLinesStdOut` and friends could keep the elements unchanged
Currently `showLinesStdOut` uses `evalMap` to print each element, discarding its value. I think these operators would be much more useful if they kept the original elements (so that there's a nice and easy way to debug the values of a stream at a stage without having to reimplement `showLinesStdOut` with `evalTap`).

### PR DESCRIPTION
**Pull request by mpilquist**
Fixes #1569

### CODE REVIEW

**Comment by mpilquist** [comment]
@kubukoz How about this?

File: core/shared/src/main/scala/fs2/Stream.scala [hunk 1]
```diff
@@ -656,6 +656,69 @@ final class Stream[+F[_], +O] private (private val free: FreeC[Nothing, O, Unit]
...
+    * @example {{{
+    * scala> Stream(1, 2).append(Stream(3, 4)).debug("a").toList
+    * a: 1
+    * a: 2
+    * a: 3
+    * a: 4
+    * res0: List[Int] = List(1, 2, 3, 4)
+    * }}}
+    */
+  def debug(tag: String = "", logger: String => Unit = println(_)): Stream[F, O] =
```
Comment (kubukoz) [inline core/shared/src/main/scala/fs2/Stream.scala]: Maybe unsafeLogger?

**Comment by kubukoz** [comment]
Looks very nice, I'll see what it looks like in some examples :) Although I'm starting to get worried about having so many methods (4 of them being for debugging). Maybe it's a good time to think about putting them in some importable object (as pipes), or an implicit class...

**Comment by mpilquist** [comment]
Yeah, it's a tricky balance. I could see getting rid of the format varieties, leaving two methods. And moving them to somewhere else kind of defeats the point of having a quick / easily accessible way of debugging.

File: core/shared/src/main/scala/fs2/Stream.scala (hunk 1 above)
Comment (mpilquist) [reply core/shared/src/main/scala/fs2/Stream.scala]: Hm, the call site seems clunkier then -- `s.debug(unsafeLogger = println)`

Comment (kubukoz) [reply core/shared/src/main/scala/fs2/Stream.scala]: Just thinking about side effects in the API. Arguably `debug` itself suggests that there will be something shady going on (e.g. `toString` for formatting),  so maybe `logger` will also be an acceptable trade-off :)

**Comment by kubukoz** [comment]
Good point, it's probably a case of something you want to access quickly... but the API growing might eventually be a problem so we can think about it next time when adding new combinators :) 

Just checked the new functions, everything seemed fine - 👍 from me

**Comment by SystemFw** [comment]
How about only leaving the `format*` signatures instead? `_ => "myTag"` does not seem very problematic if `O` infers

**Comment by kubukoz** [comment]
@SystemFw it's more like `"myTag: " + _`, but fair enough

**Pull request merged**